stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/compiler/WebSocketResourceValidator.java

```diff
@@ -73,7 +73,7 @@ private static void validateOnTextResource(String serviceName, BLangResource res
         List<BLangVariable> paramDetails = resource.getParameters();
         validateParamDetailsSize(paramDetails, 2, 3, serviceName, resource, dlog);
         validateEndpointParameter(serviceName, resource, dlog, paramDetails);
-        if (!"string".equals(paramDetails.get(1).type.toString())) {
+        if (paramDetails.size() < 2 || !"string".equals(paramDetails.get(1).type.toString())) {
```

Comment:
The following sort of type checking needs to be avoided: `"string".equals(paramDetails.get(1).type.toString())`

Reply:
How should it be checked?

Reply:
Yes, should be changed in `ResourceSignatureValidator` as well. There's an API for type checking. We need to get/construct the types of the params and use that for validation, instead of doing string comparison